# 49 Feature/conventional commits

Lucas-Dreveck opened the pull request:
Closes #49 

## 📑 Description
Adicionado conventional commits para padronizar a forma que são feitos os commits e commitzen para facilitar a aderência a essa nova funcionalidade junto a um novo handler do husky onde está sendo feito a validação automática com o commit lint

Lucas-Dreveck committed:
chore: add commitlint, commitizen and husky to commit pattern integration

Lucas-Dreveck committed:
feat: added commitlint config file and guide for the new commit style

Config commitlint, ignore the file in the esling and added guide to this new method into the README
and CONTRIBUTING.md for better team integration and padronization

feat #49

Lucas-Dreveck committed:
ci: add husky commit-msg handler

add the last feature about conventional commits to the husky commit-msg file for auto-validation

Lucas-Dreveck committed:
Merge branch 'development' into feature/conventional-commits

GusDechotti commented:
@Lucas-Dreveck eu ia arrumar os conflitos mas não tenho certeza do que é certo, pode dar uma olhadinha para que eu possa mergear?
Assim que fizer me manda uma mensagem no zap

Lucas-Dreveck committed:
Merge branch 'development' into feature/conventional-commits with recreating of the package-lock file

Lucas-Dreveck commented:
> @Lucas-Dreveck eu ia arrumar os conflitos mas não tenho certeza do que é certo, pode dar uma olhadinha para que eu possa mergear? Assim que fizer me manda uma mensagem no zap

Finalizei a solução dos conflitos, devido ao grande número de pacotes adicionados desde a ultima vez que foi criado o arquivo `package-lock.json` eu resolvi apagar e recriar ele para seguir como o próprio npm sugere

Lucas-Dreveck committed:
fix(server): changed expiresIn auth token from string|number to number

changed the type of the `expiresIn` auth token varible because of lib version changes where it needs
to be i number in seconds

Lucas-Dreveck committed:
refactor: removed unused `.env.example` from server and updated in the root

removed unused `.env.example` from server and updated the `.env.example` at the root with the
missing variables from the server workspace

Lucas-Dreveck committed:
chore: fixed incorrect dependecies location between workspaces

move `devDependecies` to the root `package.json` and `dependencies` to the correct workspace because
of the package otimization by workspace
